src/c4/C4_MPI.cc

```diff
@@ -148,6 +156,8 @@ void blocking_probe(int source, int tag, int &message_size) {
 
 //---------------------------------------------------------------------------//
 void wait_all(int count, C4_Req *requests) {
+  Require(count >= 0);
```

Comment:
Could the argument type for count be changed to unsigned, to make this precondition check unnecessary?